run_querytns.py

```diff
@@ -0,0 +1,225 @@
+
+# Set logging level:
+logging_level = logging.INFO
+verbose = True
+if args.quiet:
+    logging_level = logging.WARNING
+    verbose = False
+elif args.debug:
+    logging_level = logging.DEBUG
+    verbose = True
+
+# Setup logging:
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+console = logging.StreamHandler()
+console.setFormatter(formatter)
+logger = logging.getLogger(__name__)
+if not logger.hasHandlers():
+    logger.addHandler(console)
+logger.setLevel(logging_level)
+
+# API key for Bot
+config = load_config()
+api_key = config.get('TNS', 'api_key')
+# PARAMETERS
+days_before_today = args.days_begin  # end = now - dbt
+days_to_include = args.days_end  # begin = now - dti
+z_min = args.zmin
+z_max = args.zmax
+months = args.limit_months  # pre-limit TNS search to candidates reported in the last X months
+if np.floor(days_before_today/30) > months:
+    logger.warning('Months limit restricts days_begin, consider increasing limit_months.')
+objtype = args.objtype  # Relevant TNS SN Ia subtypes.
+
+#--------------------------------------------------------------------
+# TNS API FUNCTIONS
+# Pre-provided helper functions for the TNS API
+# Obtained from https://wis-tns.weizmann.ac.il/content/tns-getting-started
+# function for changing data to json format
+url_tns_api = "https://wis-tns.weizmann.ac.il/api/get"
+def format_to_json(source):
+    # change data to json format and return
+    parsed = json.loads(source,object_pairs_hook=OrderedDict)
+    result = json.dumps(parsed,indent=4)
+    return result
+
+# function for search obj
+def search(url,json_list):
+    try:
+        # url for search obj
+        search_url = url+'/search'
+        # change json_list to json format
+        json_file = OrderedDict(json_list)
+        # construct the list of (key,value) pairs
+        search_data = [('api_key',(None, api_key)),
+                     ('data',(None,json.dumps(json_file)))]
+        # search obj using request module
+        response = requests.post(search_url, files=search_data)
+        # return response
+        return response
+    except Exception as e:
+        return [None,'Error message : \n'+str(e)]
+
+# function for get obj
+def get(url,json_list):
+    try:
+        # url for get obj
+        get_url = url+'/object'
+        # change json_list to json format
+        json_file = OrderedDict(json_list)
+        # construct the list of (key,value) pairs
+        get_data = [('api_key',(None, api_key)),
+                     ('data',(None,json.dumps(json_file)))]
+        # get obj using request module
+        response = requests.post(get_url, files=get_data)
+        # return response
+        return response
+    except Exception as e:
+        return [None,'Error message : \n'+str(e)]
+#--------------------------------------------------------------------
+#CALCULATE CURRENT DATE AND DATE RANGE TO SEARCH
+dt_end = datetime.date(datetime.now()) - timedelta(days=days_before_today)
```

Comment:
Which timezone? Is TNS using UTC, in which case you need to be careful since datetime.now returns time in local timezone.

Reply:
Changed to utc time